### ISSUE TITLE
expand tilde to home in pathnames

### ISSUE THREAD

**Issue by daxim** [ugexe/zef#161]
`--install-to=~/local/share/perl6` treats the tilde as literal character, making it a pathname relative to the current directory. That's running counter to rule of least surprise. To fix this, either

* conform to [getopt_long](http://man7.org/linux/man-pages/man3/getopt.3.html#getopt_long%28%29%20and%20getopt_long_only%28%29) and allow options to be supplied in the form without equal character, thus: `--install-to ~/local/share/perl6` so that the shell takes care of expanding the tilde
* or call [glob(3)](http://man7.org/linux/man-pages/man3/glob.3.html) on the pathname in order to expand the tilde

**Comment by ugexe** [comment]
Maybe you should file a bug about least surprise to rakudo about issues you have with tilde expansion.

**Comment by Leont** [comment]
Having a sensible implementation of ~/ should be trivial. Replacing ~/ with `%*ENV<HOME>` on uix, and `%*<USER_PROFILE>` on Windows should do the trick.

**Comment by AlexDaniel** [comment]
Wait, why? `~` is a thing in shells, zef is not a shell. If I wanted to install into `~` directory (yes, directory named `~`) then I'd expect zef to do what I told it to do, not to reinterpret the path in weird shell ways.

**Comment by tony-o** [comment]
++@alexdaniel, that sounds like an upstream feature request

**Comment by AlexDaniel** [comment]
@Leont but you can use `$HOME` env variable, and you can even hit `C-M-e` in bash to expand it right away.

Zef should not mangle arguments. I doubt *any* proper tool in existence does that.

**Comment by Leont** [comment]
> Zef should not mangle arguments. I doubt any proper tool in existence does that.

*Proper* makes it a [no true scotsman](https://en.wikipedia.org/wiki/No_true_Scotsman) argument, but I have maintained tools that did this (e.g. [Module::Build](https://metacpan.org/search?q=Module%3A%3ABuild)

**Comment by AlexDaniel** [comment]
> Proper makes it a no true scotsman argument, but I have maintained tools that did this

Correct. But I'm more interested in seeing their justification (if there is any, which I originally doubted, so that we don't have to do bikeshedding ourselves) than playing fallacy games. [From the ticket for `Module::Build`](https://rt.cpan.org/Public/Bug/Display.html?id=13381):

> perl Build.PL --install_base=~ causes everything to be installed into a subdirectory literally called '~'. Even though this is not technically Module::Build's fault, its the shell, its very surprising and very dangerous as a user will instinctively do "rm -rf ~" and blow away their home directory. I nearly did so.

That's interesting. I don't know if `--install-to=~` reasoning can be applied here though.
